framework/src/media/InputHandler.h

```diff
@@ -72,7 +75,11 @@ class InputHandler : public StreamHandler
 	ssize_t getElementaryStream(unsigned char *buf, size_t size, size_t *used, unsigned char **out, size_t *expect);
 	ssize_t getPCM(unsigned char *buf, size_t size, size_t *used, unsigned char **out, size_t *expect);
 	size_t fetchData(unsigned char *buf, size_t size, size_t *used, unsigned char **out, size_t *expect);
+	ssize_t readFromSource(unsigned char *buf, size_t size);
 
+	std::mutex mMutex;
```

Comment:
should you add these new members to the initializing list?

Reply:
std::mutex has default constructor, and the mutex is in unlocked state after the constructor completes.
We don't need additional explicit initialization.
std::condition_variable and std::shared_ptr<...> is similar.